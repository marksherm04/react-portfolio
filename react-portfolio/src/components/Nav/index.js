import React from 'react';
import { capitalizeFirstLetter } from "../../utils/helpers";


function Nav() {

	const categories = [
		{ name: 'portfolio', description: 'Photos of different projects of mine' }
	];

	const handleClick = () => {
		console.log("click handled");
	};

	return (
		<header className="flex-row px-1">
			<h1>
				<a data-testid="link">
					<span role="img"></span> Mark Sherman
				</a>
			</h1>
			<nav>
				<ul className="flex-row">
					<li className="mx-2">
						<a data-testid="about" href="#about" onClick={() => handleClick()}>
							About Me
						</a>
					</li>
					<li className={"mx-2"}>
						<a data-testid="contact" href="#contact" onClick={() => handleClick()}>
							Contact
						</a>
					</li>
					{categories.map((category) => (
						<li className="mx-1" key={category.name} >
							<a href="#portfolio" onClick={() => { handleClick(); }}>
								{capitalizeFirstLetter(category.name)}
							</a>
						</li>
					))}
				</ul>
			</nav>
		</header>
	);
}

export default Nav;