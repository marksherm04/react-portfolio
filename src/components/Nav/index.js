import React from 'react';

function Nav(props) {
	 const { currentLink, setCurrentLink } = props;
	
	 return (
		 <nav>
			 <ul className="text-2xl font-bold text-blue-500 flex py-10 px-4 font-serif cursor-pointer static">
				 <li className={currentLink === "about" ? "rounded bg-blue-200 hover:bg-yellow-300 navActive" : "m-8 md:my-8"}>
					 <span onClick={() => setCurrentLink('about')}>About Me</span>
				 </li>
				 <li className={currentLink === "portfolio" ? "rounded bg-blue-200 hover:bg-yellow-300 navActive" : "m-8 md:my-8"}>
					 <span onClick={() => setCurrentLink("portfolio")}>Portfolio</span>
				 </li>
				 <li className={currentLink === "contact" ? "rounded bg-blue-200 hover:bg-yellow-300 navActive" : "m-8 md:my-8"}>
					 <span onClick={() => setCurrentLink("contact")}>Contact</span>
				 </li>
				 <li className={currentLink === "resume" ? "rounded bg-blue-200 hover:bg-yellow-300 navActive" : "m-8 md:my-8"}>
					 <span onClick={() => setCurrentLink("resume")}>Resume</span>
				 </li>
				 <li className={currentLink === "name" ? "navActive" : "m-8"}>
					 <span onClick={() => setCurrentLink("about")}>Mark Sherman 👨‍💻</span>
				 </li>
			 </ul>
		 </nav>
	 );
}

export default Nav;