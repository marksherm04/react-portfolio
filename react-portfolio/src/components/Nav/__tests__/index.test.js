import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

describe('Nav component', () => {
	// baseline test
	it('renders', () => {
		render(<Nav />);
	});

	// snapshot test
	it('matches snapshot', () => {
		const { asFragment } = render(<Nav />);
		// assert value comparison
		expect(asFragment()).toMatchSnapshot();
	});
})

describe('links are visible', () => {
	it('inserts text into the links', () => {
		// Arrange
		const { getByTestId } = render(<Nav />);
		// Assert
		expect(getByTestId('link')).toHaveTextContent('Mark Sherman');
		expect(getByTestId('about')).toHaveTextContent('About Me');
		expect(getByTestId('contact')).toHaveTextContent('Contact');
	});
})