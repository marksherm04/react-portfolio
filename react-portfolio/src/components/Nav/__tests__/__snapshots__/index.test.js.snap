// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1"
  >
    <h1>
      <a
        data-testid="link"
        href="/"
      >
        <span
          role="img"
        />
         Mark Sherman
      </a>
    </h1>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About Me
          </a>
        </li>
        <li
          class="mx-2"
        >
          <a
            data-testid="contact"
            href="#contact"
          >
            Contact
          </a>
        </li>
        <li
          class="mx-1"
        >
          <a
            href="#projects"
          >
            Projects
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
